app/cdap/components/Connections/Browser/GenericBrowser/index.tsx

```diff
@@ -166,6 +166,7 @@ export function GenericBrowser({ selectedConnection }) {
       const plugin = spec?.relatedPlugins?.[0];
 
       const properties = plugin?.properties;
+      properties.referenceName = plugin.name;
```

Comment:
- Correctness: The other accesses to `plugin` around this line are using optional chaining, so you likely also need to use optional chaining, and properties might be null. I'm surprised and a bit disappointed in typescript that it didn't infer this and warn you, actually. (Perhaps it simply believes the optional chaining is unnecessary in the first place, though, haha.)
- Question: Can this ever overwrite a user's input of referenceName? is GenericBrowser ever used to navigate configured nodes in a pipeline, for example?